### PR TITLE
smuxi: init at unstable-2023-07-01

### PR DESCRIPTION
**Pull request by meebey**
## Description of changes

* Adding Smuxi that I am also the author of.
* GTK# is a dependency and was restricted to Linux even though it builds and works fine on Darwin so I adjusted the platform line for it.
* Added myself as maintainer.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by stigtsp** [comment]
@GrahamcOfBorg eval

File: pkgs/applications/networking/instant-messengers/smuxi/default.nix [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+, fetchFromGitHub
+, glib
+, gettext
+, mono
+, stfl
+, makeWrapper, lib
+, guiSupport ? true
+, gtk-sharp-2_0
+, gdk-pixbuf
+, gnome2
```
Comment (stigtsp) [inline pkgs/applications/networking/instant-messengers/smuxi/default.nix:12]:
```suggestion
, pango
```

File: pkgs/applications/networking/instant-messengers/smuxi/default.nix [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                                                  gettext stfl
+                                                 ]}
+
+    makeWrapper "${mono}/bin/mono" "$out/bin/smuxi-frontend-gnome" \
+      --add-flags "$out/lib/smuxi/smuxi-frontend-gnome.exe" \
+      --prefix MONO_GAC_PREFIX : ${if guiSupport then gtk-sharp-2_0 else ""} \
+      --prefix ${runtimeLoaderEnvVariableName} : ${lib.makeLibraryPath [
+                                                   gettext
+                                                   glib
+                                                   gtk-sharp-2_0 gtk-sharp-2_0.gtk gdk-pixbuf gnome2.pango
```
Comment (stigtsp) [inline pkgs/applications/networking/instant-messengers/smuxi/default.nix:76]:
```suggestion
                                                   gtk-sharp-2_0 gtk-sharp-2_0.gtk gdk-pixbuf pango
```

File: pkgs/applications/networking/instant-messengers/smuxi/default.nix [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ autoconf automake itstool intltool gettext
+    mono
+    stfl
+    makeWrapper ] ++ lib.optionals (guiSupport) [
+      gtk-sharp-2_0
+      # loaded at runtime by GTK#
+      gdk-pixbuf gnome2.pango
```
Comment (stigtsp) [inline pkgs/applications/networking/instant-messengers/smuxi/default.nix:39]:
```suggestion
      gdk-pixbuf pango
```

**Review by stigtsp — COMMENTED** [review]
Replacing `gnome2.pango` with `pango` makes nix-review happy on my end, but not sure if this is the right way to go?

**Comment by meebey** [comment]
> Replacing `gnome2.pango` with `pango` makes nix-review happy on my end, but not sure if this is the right way to go?

looks good to me, it builds and works, so I committed and pushed the change

**Comment by stigtsp** [comment]
> > Replacing `gnome2.pango` with `pango` makes nix-review happy on my end, but not sure if this is the right way to go?
> 
> @stigtsp looks good to me, it builds and works, so I committed and pushed the change

Could you amend it into the `smuxi: init at unstable-2023-07-01` commit?

**Comment by meebey** [comment]
> > > Replacing `gnome2.pango` with `pango` makes nix-review happy on my end, but not sure if this is the right way to go?
> > 
> > 
> > @stigtsp looks good to me, it builds and works, so I committed and pushed the change
> 
> Could you amend it into the `smuxi: init at unstable-2023-07-01` commit?

I could squash the commit but then the issue and solution won't be able to other people who may also run into it.

**Comment by meebey** [comment]
I have resolved the maintainer-list.nix conflict

**Comment by meebey** [comment]
Squashed and rebased, should be good as is to be merged @stigtsp

**Review by stigtsp — APPROVED** [review]
LGTM, thanks!

**Pull request merged**